api/qan/metrics.proto

```diff
@@ -6,54 +6,58 @@ option go_package = "qan";
 
 import "google/api/annotations.proto";
 
-// Metrics TODO.
+// Metrics serves agregated metrics filtered by given dimention value and period.
 service Metrics {
-  // GetMetricsByDigest TODO.
-  rpc GetMetricsByDigest(MetricsRequest) returns (MetricsReply) {
+  // GetMetrics gets map of metrics for specific filtering.
+  rpc GetMetrics(MetricsRequest) returns (MetricsReply) {
     option (google.api.http) = {
-      post: "/v1/qan/GetMetricsByDigest"
+      post: "/v1/qan/GetMetrics"
       body: "*"
     };
   }
 }
 
+// MetricsRequest defines filtering of metrics for specific value of dimention (ex.: host=hostname1 or queryid=1D410B4BE5060972.
 message MetricsRequest {
-  string from = 1;
-  string to = 2;
-  string digest = 3;
-  repeated MapFieldEntry labels = 4;
-  repeated string include = 5;
+  string period_start_from = 1;
+  string period_start_to = 2;
+  string filter_by = 3; // dimention: queryid | host ...
+  string group_by = 4; // one of dimention
+  repeated MapFieldEntry labels = 5;
+  repeated string include_only_fields = 6;
 }
 
-// MapFieldEntry TODO.
+// MapFieldEntry allows to pass labels/dimentions in form like {"d_server": ["db1", "db2"...]}.
 message MapFieldEntry {
   string key = 1;
   repeated string value = 2;
 }
 
+// MetricsReply defines metrics for specific value of dimention (ex.: host=hostname1 or queryid=1D410B4BE5060972.
 message MetricsReply {
-  string digest = 2; // md5 of digest_text
-  string digest_text = 3; // digest_text - query signature. Query without values.
+  string queryid = 1; // md5 of digest_text
+  string fingerprint = 2; // digest_text - query signature. Query without values.
   // Dimension Group.
-  repeated string db_servers = 4;
-  repeated string db_schemas = 5;
-  repeated string db_usernames = 6;
-  repeated string client_hosts = 7;
-  string first_seen = 201;
-  string last_seen = 202;
-  map<string, string> labels = 8;
-  string agent_uuid = 9;
-  int64 period_start = 10;
-  uint32 period_length = 11;
-  string example = 12;
+  repeated string d_servers = 3;
+  repeated string d_databases = 4;
+  repeated string d_schemas = 5;
+  repeated string d_usernames = 6;
+  repeated string d_client_hosts = 7;
+  string first_seen = 8;
+  string last_seen = 9;
+  map<string, string> labels = 10;
+  string agent_uuid = 11;
+  int64 period_start = 12;
+  uint32 period_length = 13;
+  string example = 14;
   // ExampleFormat is formant of query example: real or query without values
   enum ExampleFormat {
```

Comment:
Duplicate of https://github.com/percona/pmm/pull/268/files#diff-283a55e2672ed0140ef30d059c3be9e0R37.

Reply:
Moved to qan.proto